### ISSUE TITLE
Adapt for ADC resolution on ESP32

### ISSUE THREAD

**Issue by lukasfischer83** [adafruit/Adafruit_TouchScreen#15]
Arduino board:  ESP32
Hi, I just noticed a small bug on ESP32, which has 12bit ADC resolution. The driver works as expected if   `analogReadResolution(10);`  is called before using `getPoint()`. Otherwhise, as expected, it does not ;). 
So maybe you could implement a check on the resolution and change the calculations (1023 is hardcoded in some lines) or simply right-shift the adc results by two if ESP32 is defined.
Cheers, Lukas

**Comment by Ryan-obl** [comment]
Good catch, thank goodness I found your post (and your simple solution) in time!

Now the Adafruit touchscreen works like a charm on my ESP32!

**Comment by LeonidKokhnovich** [comment]
@lukasfischer83 Hey! I wonder what pins do you use on ESP32 for Xp, Xm, Yp and Ym? It doesn't seem to work for me.

**Comment by Psycho-0** [comment]
it would appear that analogReadResolution(10) does not work on the NodeMCU ESP32s model. When given an (11) or (12) value it functions but only yields constant values for X, Y location. However, pressure does work as expected.

**Comment by Ryan-obl** [comment]
> @lukasfischer83 @Ryan-obl  Hey! I wonder what pins do you use on ESP32 for Xp, Xm, Yp and Ym? It doesn't seem to work for me.

Hey @LeonidKokhnovich , I would use any pins that has any of the ESP32's ADCs on it, which naturally would depend on which board your ESP32 is on. 

For instance, I'm using the ESP32 DevKitC v4, whose pinout can be found at:
[https://components101.com/microcontrollers/esp32-devkitc](url)

Cheers, Ryan

**Comment by hotteshen** [comment]
> Arduino board: ESP32
> Hi, I just noticed a small bug on ESP32, which has 12bit ADC resolution. The driver works as expected if `analogReadResolution(10);` is called before using `getPoint()`. Otherwhise, as expected, it does not ;).
> So maybe you could implement a check on the resolution and change the calculations (1023 is hardcoded in some lines) or simply right-shift the adc results by two if ESP32 is defined.
> Cheers, Lukas

This was so much helpful.
I struggled with ESP32 touchscreen issue.
The touchpoint values were strange. I thought some other circuits are interfering electric signals and so ESP32 is reading wrong values. But it was ADC resolution issue. I realized that after reading this.
Thanks again!

**Comment by ScruffR** [comment]
I just filed a pull-request to solve that problem
https://github.com/adafruit/Adafruit_TouchScreen/pull/43
